File: Components/WinForms/CS/ChartsMainDemo/Modules/RadarPolarViews/ChartDemoRadarPolarBase.cs
using System;
using DevExpress.XtraEditors;

namespace DevExpress.XtraCharts.Demos.RadarPolarViews {
    public partial class RadarPolarDemoBase : ChartDemoBase2D {
        LabelControl labelMarkerKind;
        LabelControl labelMarkerSize;
        MarkerKindItemCollection itemCollection;

        protected virtual bool ShowMarkers { get { return true; } set { } }
        protected virtual bool ShowDiagramStyle { get { return true; } }
        protected override bool SeriesSelection { get { return true; } }
        
        public RadarPolarDemoBase() {
            InitializeComponent();
            labelDiagramStyle.Visible = ShowDiagramStyle;
            comboBoxEditDiagramStyle.Visible = ShowDiagramStyle;
        }
        
        void comboBoxEditMarkerKind_SelectedIndexChanged(object sender, EventArgs e) {
            if (this.comboBoxEditMarkerKind.SelectedIndex == -1 || seriesSelected == null)
                return;
            RadarPointSeriesView view = seriesSelected.View as RadarPointSeriesView;
            if (view != null) {
                MarkerKindItem item = (MarkerKindItem)this.comboBoxEditMarkerKind.SelectedItem;
                view.PointMarkerOptions.Kind = item.MarkerKind;
                if (item.MarkerKind.Equals(MarkerKind.Star))
                    view.PointMarkerOptions.StarPointCount = ((StarMarkerKindItem)item).PointCount;
            }
        }
        void comboBoxEditMarkerSize_SelectedIndexChanged(object sender, EventArgs e) {
            if (this.comboBoxEditMarkerKind.SelectedIndex == -1 || this.seriesSelected == null)
                return;
            RadarPointSeriesView view = seriesSelected.View as RadarPointSeriesView;
            if (view != null)
                view.PointMarkerOptions.Size = Int32.Parse(this.comboBoxEditMarkerSize.Text);
        }
        void comboBoxEditDiagramStyle_SelectedIndexChanged(object sender, EventArgs e) {
            RadarDiagram diagram = ChartControl.Diagram as RadarDiagram;
            if (diagram != null)
                diagram.DrawingStyle = (RadarDiagramDrawingStyle)comboBoxEditDiagramStyle.SelectedIndex;
        }
        
        protected override void InitControls() {
            base.InitControls();
            this.itemCollection = MarkerKindItemCollection.CreateCollection();
            MarkerKindItem[] items = new MarkerKindItem[this.itemCollection.Count];
            this.itemCollection.CopyTo(items);
            this.comboBoxEditMarkerKind.Properties.Items.AddRange(items);
            if (ChartControl.Series.Count > 0)
                ShowLabels = ChartControl.Series[0].LabelsVisibility == DevExpress.Utils.DefaultBoolean.True;
            RadarDiagram diagram = ChartControl.Diagram as RadarDiagram;
            if (diagram != null)
                comboBoxEditDiagramStyle.SelectedIndex = (int)diagram.DrawingStyle;
            this.comboBoxEditMarkerSize.Properties.Items.AddRange(MarkerKindItemCollection.GetSizeArray());
        }
        protected void EnableMarkerControls(bool enable) {
            this.labelMarkerKind.Enabled = enable;
            this.comboBoxEditMarkerKind.Enabled = enable;
            this.labelMarkerSize.Enabled = enable;
            this.comboBoxEditMarkerSize.Enabled = enable;
        }

        public override void UpdateControls() {
            base.UpdateControls();
            if (this.seriesSelected == null) {
                this.comboBoxEditMarkerKind.SelectedIndex = -1;
                this.comboBoxEditMarkerKind.Enabled = false;
                this.labelMarkerKind.Enabled = false;
                this.comboBoxEditMarkerSize.SelectedIndex = -1;
                this.comboBoxEditMarkerSize.Enabled = false;
                this.labelMarkerSize.Enabled = false;
            }
            else {
                RadarPointSeriesView view = seriesSelected.View as RadarPointSeriesView;
                if (view != null) {
                    if (view.PointMarkerOptions.Kind != MarkerKind.Star)
                        this.comboBoxEditMarkerKind.SelectedItem = this.itemCollection.GetItemByKind(view.PointMarkerOptions.Kind);
                    else
                        this.comboBoxEditMarkerKind.SelectedItem = this.itemCollection.GetItemByStarPointCount(view.PointMarkerOptions.StarPointCount);
                    this.comboBoxEditMarkerKind.Enabled = ShowMarkers;
                    this.labelMarkerKind.Enabled = ShowMarkers;
                    this.comboBoxEditMarkerSize.Enabled = ShowMarkers;
                    this.labelMarkerSize.Enabled = ShowMarkers;
                    this.comboBoxEditMarkerSize.Text = view.PointMarkerOptions.Size.ToString();
                }
            }
        }
    }
}

File: Components/WinForms/VB/ChartsMainDemo/Modules/RadarPolarViews/ChartDemoRadarPolarBase.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors

Namespace DevExpress.XtraCharts.Demos.RadarPolarViews
	Partial Public Class RadarPolarDemoBase
		Inherits ChartDemoBase2D
		Private labelMarkerKind As LabelControl
		Private labelMarkerSize As LabelControl
		Private itemCollection As MarkerKindItemCollection

		Protected Overridable Property ShowMarkers() As Boolean
			Get
				Return True
			End Get
			Set(ByVal value As Boolean)
			End Set
		End Property
		Protected Overridable ReadOnly Property ShowDiagramStyle() As Boolean
			Get
				Return True
			End Get
		End Property
		Protected Overrides ReadOnly Property SeriesSelection() As Boolean
			Get
				Return True
			End Get
		End Property

		Public Sub New()
			InitializeComponent()
			labelDiagramStyle.Visible = ShowDiagramStyle
			comboBoxEditDiagramStyle.Visible = ShowDiagramStyle
		End Sub

		Private Sub comboBoxEditMarkerKind_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxEditMarkerKind.SelectedIndexChanged
			If Me.comboBoxEditMarkerKind.SelectedIndex = -1 OrElse seriesSelected Is Nothing Then
				Return
			End If
			Dim view As RadarPointSeriesView = TryCast(seriesSelected.View, RadarPointSeriesView)
			If view IsNot Nothing Then
				Dim item As MarkerKindItem = CType(Me.comboBoxEditMarkerKind.SelectedItem, MarkerKindItem)
				view.PointMarkerOptions.Kind = item.MarkerKind
				If item.MarkerKind.Equals(MarkerKind.Star) Then
					view.PointMarkerOptions.StarPointCount = (CType(item, StarMarkerKindItem)).PointCount
				End If
			End If
		End Sub
		Private Sub comboBoxEditMarkerSize_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxEditMarkerSize.SelectedIndexChanged
			If Me.comboBoxEditMarkerKind.SelectedIndex = -1 OrElse Me.seriesSelected Is Nothing Then
				Return
			End If
			Dim view As RadarPointSeriesView = TryCast(seriesSelected.View, RadarPointSeriesView)
			If view IsNot Nothing Then
				view.PointMarkerOptions.Size = Int32.Parse(Me.comboBoxEditMarkerSize.Text)
			End If
		End Sub
		Private Sub comboBoxEditDiagramStyle_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxEditDiagramStyle.SelectedIndexChanged
			Dim diagram As RadarDiagram = TryCast(ChartControl.Diagram, RadarDiagram)
			If diagram IsNot Nothing Then
				diagram.DrawingStyle = CType(comboBoxEditDiagramStyle.SelectedIndex, RadarDiagramDrawingStyle)
			End If
		End Sub

		Protected Overrides Sub InitControls()
			MyBase.InitControls()
			Me.itemCollection = MarkerKindItemCollection.CreateCollection()
			Dim items(Me.itemCollection.Count - 1) As MarkerKindItem
			Me.itemCollection.CopyTo(items)
			Me.comboBoxEditMarkerKind.Properties.Items.AddRange(items)
			If ChartControl.Series.Count > 0 Then
				ShowLabels = ChartControl.Series(0).LabelsVisibility = DevExpress.Utils.DefaultBoolean.True
			End If
			Dim diagram As RadarDiagram = TryCast(ChartControl.Diagram, RadarDiagram)
			If diagram IsNot Nothing Then
				comboBoxEditDiagramStyle.SelectedIndex = CInt(Fix(diagram.DrawingStyle))
			End If
			Me.comboBoxEditMarkerSize.Properties.Items.AddRange(MarkerKindItemCollection.GetSizeArray())
		End Sub
		Protected Sub EnableMarkerControls(ByVal enable As Boolean)
			Me.labelMarkerKind.Enabled = enable
			Me.comboBoxEditMarkerKind.Enabled = enable
			Me.labelMarkerSize.Enabled = enable
			Me.comboBoxEditMarkerSize.Enabled = enable
		End Sub

		Public Overrides Sub UpdateControls()
			MyBase.UpdateControls()
			If Me.seriesSelected Is Nothing Then
				Me.comboBoxEditMarkerKind.SelectedIndex = -1
				Me.comboBoxEditMarkerKind.Enabled = False
				Me.labelMarkerKind.Enabled = False
				Me.comboBoxEditMarkerSize.SelectedIndex = -1
				Me.comboBoxEditMarkerSize.Enabled = False
				Me.labelMarkerSize.Enabled = False
			Else
				Dim view As RadarPointSeriesView = TryCast(seriesSelected.View, RadarPointSeriesView)
				If view IsNot Nothing Then
					If view.PointMarkerOptions.Kind <> MarkerKind.Star Then
						Me.comboBoxEditMarkerKind.SelectedItem = Me.itemCollection.GetItemByKind(view.PointMarkerOptions.Kind)
					Else
						Me.comboBoxEditMarkerKind.SelectedItem = Me.itemCollection.GetItemByStarPointCount(view.PointMarkerOptions.StarPointCount)
					End If
					Me.comboBoxEditMarkerKind.Enabled = ShowMarkers
					Me.labelMarkerKind.Enabled = ShowMarkers
					Me.comboBoxEditMarkerSize.Enabled = ShowMarkers
					Me.labelMarkerSize.Enabled = ShowMarkers
					Me.comboBoxEditMarkerSize.Text = view.PointMarkerOptions.Size.ToString()
				End If
			End If
		End Sub
	End Class
End Namespace
